chains/evm/listener/event-handler.go

```diff
@@ -82,20 +83,28 @@ func Erc20EventHandler(sourceID, destId uint8, nonce uint64, resourceID types.Re
 	// within ERC20MessageHandler
 	// https://github.com/ChainSafe/chainbridge-core/blob/main/chains/evm/voter/message-handler.go#L108
 
-	// recipientAddress: last 20 bytes of calldata
-	recipientAddress := calldata[64:]
-
-	return &message.Message{
-		Source:       sourceID,
-		Destination:  destId,
-		DepositNonce: nonce,
-		ResourceId:   resourceID,
-		Type:         message.FungibleTransfer,
-		Payload: []interface{}{
-			amount,
-			recipientAddress,
-		},
-	}, nil
+	// 32-64 is recipient address length
+	recipientAddressLength := big.NewInt(0).SetBytes(calldata[32:64])
+
+	// 64 - (64 + recipient address length) is recipient address
+	recipientAddress := calldata[64:(64 + recipientAddressLength.Int64())]
+
+	// if there is priority data, parse it and use it
+	payload := []interface{}{
+		amount,
+		recipientAddress,
+	}
+
+	// arbitrary metadata that will be most likely be used by the relayer
+	var metadata []interface{}
```

Comment:
It does seem to me we can create a struct for metadata and type it correctly instead of just creating an interface as we know types of metadata we can expect.

Reply:
Since we are explicitly parsing Priority downside in this function I do not see any reason for doing this as an interface... We are 100% sure what are we parsing right now.

Reply:
So i agree with Matija